tokio/src/sync/once_cell.rs

```diff
@@ -68,11 +76,18 @@ use std::sync::atomic::{AtomicBool, Ordering};
 /// }
 /// ```
 pub struct OnceCell<T> {
-    value_set: AtomicBool,
+    state: AtomicU8,
     value: UnsafeCell<MaybeUninit<T>>,
-    semaphore: Semaphore,
+    waiters: Mutex<LinkedList<Waiter, <Waiter as linked_list::Link>::Target>>,
+    /// The current number of available permits in the semaphore.
+    #[cfg(all(tokio_unstable, feature = "tracing"))]
+    resource_span: tracing::Span,
 }
 
+const STATE_UNSET: u8 = 0;
+const STATE_LOCKED: u8 = 1;
+const STATE_SET: u8 = 2;
```

Comment:
If you swap these, then a ZST oncecell will only have states 0 and 1 instead of states 0 and 2. That seems nicer.
```suggestion
const STATE_UNSET: u8 = 0;
const STATE_LOCKED: u8 = 2;
const STATE_SET: u8 = 1;
```